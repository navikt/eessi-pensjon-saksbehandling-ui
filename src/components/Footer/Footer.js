import React, { useState } from 'react'
import PT from 'prop-types'
import classNames from 'classnames'
import { Nav } from 'eessi-pensjon-ui'
import './Footer.css'

const Footer = ({ actions, footerOpen, params = {} }) => {
  const validParams = ['buc', 'sed', 'rinaId', 'sakId', 'aktoerId', 'avdodfnr', 'vedtakId', 'kravId', 'fnr', 'mottaker']
  const [paramName, setParamName] = useState(undefined)
  const [paramValue, setParamValue] = useState(undefined)

  const onUnsetParam = (key) => {
    actions.unsetStatusParam(key)
  }

  const onSetParam = () => {
    actions.setStatusParam(paramName, paramValue)
    setParamName(undefined)
    setParamValue(undefined)
  }

  const onSetParamName = (e) => {
    const name = e.target ? e.target.value : e
    if (name !== '') {
      setParamName(name)
    }
  }

  const onSetParamValue = (e) => {
    const value = e.target.value
    if (value !== '') {
      setParamValue(value)
    }
  }

  const toggleFooterOpen = () => {
    actions.toggleFooterOpen()
  }

  return (
    <footer className={classNames('c-footer', { toggled: footerOpen })}>
      <div className={classNames('contents', { fullWidth: !footerOpen })}>
        <div
          className={classNames({ footerButtonOpen: footerOpen, footerButtonClosed: !footerOpen })}
          onClick={toggleFooterOpen}
        >
          {footerOpen ? 'â–¼' : null}
        </div>
        {footerOpen
          ? (
            <div className='c-footer__form'>
              <Nav.Select id='c-footer__select-id' className='c-footer__select' label='' onChange={onSetParamName}>
                <option value=''>--</option>
                {validParams.map(param => {
                  return params[param] ? null : <option key={param} value={param}>{param}</option>
                })}
              </Nav.Select>
              <Nav.Input
                label='' id='c-footer__input-id' className='c-footer__input' value={paramValue || ''}
                onChange={onSetParamValue}
              />
              <Nav.Knapp id='c-footer__add-button-id' className='c-footer__add-button' onClick={onSetParam}>&nbsp;+&nbsp;</Nav.Knapp>
            </div>
          ) : null}
      </div>
      {footerOpen ? (
        <div className='c-footer__params'>
          {validParams.map(param => {
            return params[param] ? (
              <div key={param} className='c-footer__param'>
                <Nav.EtikettBase className='c-footer__param-string' type='info'>
                  <b>{param}</b> {params[param]}
                </Nav.EtikettBase>
                <Nav.Lukknapp className='c-footer__remove-button' onClick={() => onUnsetParam(param)} />
              </div>
            ) : null
          })}
        </div>
      ) : null}
    </footer>
  )
}

Footer.propTypes = {
  actions: PT.object.isRequired,
  footerOpen: PT.bool,
  params: PT.object
}

export default Footer
