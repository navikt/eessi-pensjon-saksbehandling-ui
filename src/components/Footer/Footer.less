@import '../../constants';

.c-footer {
  flex-shrink: 0;
  background: @navLysGra;
  padding: 0rem;
  display: flex;
  flex-direction: row;
  justify-content: space-between;
  transition: width 0.2s ease-out;

  &.toggled {
    height: 40px;
  }

  &:not(.toggled) {
    height: 10px;
  }

  .footerButtonClosed {
     width: 100%;
     margin: 0px;
     display: block;
     padding: 0px;
     &:hover {
       cursor: pointer;
       background-color: @navLysGra;
     }
  }

  .footerButtonOpen {
     width: 1.2rem;
     margin-right: 0.5rem;
     display: inline-block;
     padding-top: 0.6rem;
     color: @navGra60;
     text-align: center;
     &:hover {
       cursor: pointer;
       background-color: @navLysGra;
     }
  }
  .fullWidth {
      width: 100%;
  }
  .contents {
    display:flex;
  }
  &__form {
    transform: scale(0.7);
    transform-origin: left top;
    display: flex;
    flex-direction: row;
    align-items: baseline;
  }
  &__select,
  &__input {
     margin-right: 0.5rem;
     margin-bottom: 0px;
  }
  &__add-button {
    padding: 0.5rem;
  }
  &__params {
    display: flex;
    flex-direction: row-reverse;

    div {
      margin-left: 0.25rem;
    }
  }

  &__remove-button {
    transform: scale(0.5);
  }

  &__param {
    display: flex;
    align-items: center;
    flex-direction: row;

    .etikett {
       padding: 0.2rem !important;
       white-space: nowrap;
    }
  }
}

.highContrast.c-footer {
  background: @navMorkGra !important;
}
