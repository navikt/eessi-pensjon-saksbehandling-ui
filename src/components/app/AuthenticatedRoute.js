import React, { Component } from 'react'
import PT from 'prop-types'
import { connect } from 'react-redux'
import { bindActionCreators } from 'redux'
import { Route, withRouter, Redirect } from 'react-router'
import _ from 'lodash'

import { IS_DEVELOPMENT } from '../../constants/environment'
import WaitingPanel from './WaitingPanel'

import * as routes from '../../constants/routes'
import * as appActions from '../../actions/app'
import * as statusActions from '../../actions/status'

const mapStateToProps = (state) => {
  return {
    userRole: state.app.userRole,
    userStatus: state.app.userStatus,
    loggedIn: state.app.loggedIn,
    allowed: state.app.allowed,
    isLoggingIn: state.loading.isLoggingIn,
    gettingUserInfo: state.loading.gettingUserInfo,
    rinaId: state.status.rinaId
  }
}

const mapDispatchToProps = (dispatch) => {
  return { actions: bindActionCreators(Object.assign({}, appActions, statusActions), dispatch) }
}

const paramAliases = {
  'rinaid': 'rinaId',
  'saksNr': 'saksId',
  'fnr': 'aktoerId'
}

class AuthenticatedRoute extends Component {
  state = {
    isReady: false
  }

  parseSearchParams () {
    const { actions, location } = this.props

    let params = new URLSearchParams(location.search)
    let newParams = {}
    params.forEach((value, key) => {
      const _key = paramAliases.hasOwnProperty(key) ? paramAliases[key] : key
      const _value = value || undefined
      if (_value !== this.state[_key]) {
        actions.setStatusParam(_key, _value)
        newParams[_key] = _value
      }
    })
    if (!_.isEmpty(newParams)) {
      this.setState(newParams)
    }
    return newParams
  }

  componentDidMount () {
    const { actions, userStatus } = this.props
    if (!userStatus) {
      actions.getUserInfo()
    } else {
      this.setState({
        isReady: true
      })
    }
    this.parseSearchParams()
  }

  componentDidUpdate () {
    const { userStatus } = this.props
    const { isReady } = this.state

    if (!isReady && userStatus !== undefined) {
      this.setState({
        isReady: true
      })
    }
    this.parseSearchParams()
  }

  hasApprovedRole () {
    const { roles, userRole } = this.props
    return roles.indexOf(userRole) >= 0
  }

  comesFromPesys () {
    return this.state.hasOwnProperty('saksId') && this.state.hasOwnProperty('aktoerId')
  }

  forceLogin () {
    const { actions } = this.props

    console.log("No oidc-token, force login.")
    actions.login()
  }

  render () {
    const { userRole, allowed, gettingUserInfo } = this.props
    const { isReady } = this.state

    if (!isReady || gettingUserInfo) {
      return <WaitingPanel message='authenticating' />
    }

    if (!userRole) {
      this.forceLogin()
      return null
    }

    let validRole = this.hasApprovedRole()

    if (!validRole) {
      return <Redirect to={routes.FORBIDDEN} />
    }

    let authorized = allowed || IS_DEVELOPMENT

    if (!authorized) {
      return <Redirect to={routes.NOT_INVITED} />
    }

    return <Route {...this.props} />
  }
}

AuthenticatedRoute.propTypes = {
  className: PT.string,
  actions: PT.object.isRequired,
  roles: PT.array.isRequired
}

export default connect(
  mapStateToProps,
  mapDispatchToProps
)(
  withRouter(AuthenticatedRoute)
)
