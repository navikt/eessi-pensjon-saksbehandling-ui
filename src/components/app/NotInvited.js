import React, { Component } from 'react'
import { connect } from 'react-redux'
import PT from 'prop-types'
import { withNamespaces } from 'react-i18next'
import classNames from 'classnames'

import TopContainer from '../../components/ui/TopContainer/TopContainer'
import FrontPageDrawer from '../../components/drawer/FrontPage'

const mapStateToProps = (state) => {
  return {
    status: state.status
  }
}

class NotInvited extends Component {
  render () {
    const { t, history, location, status } = this.props

    return <TopContainer className={classNames('p-app-notInvited')}
      history={history} location={location}
      sideContent={<FrontPageDrawer t={t} status={status} />}>
      <h1 className='typo-sidetittel ml-0 appTitle'>{'not invited'}</h1>
    </TopContainer>
  }
}

NotInvited.propTypes = {
  t: PT.func.isRequired,
  history: PT.object,
  location: PT.object,
  status: PT.object
}

export default connect(
  mapStateToProps
)(
  withNamespaces()(NotInvited)
)
