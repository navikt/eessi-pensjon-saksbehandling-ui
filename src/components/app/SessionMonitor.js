import React, { Component } from 'react'
import PT from 'prop-types'
import { connect, bindActionCreators } from 'store'
import { withTranslation } from 'react-i18next'

import * as uiActions from '../../actions/ui'

const mapStateToProps = (state) => {
  return {
    loggedTime: state.app.loggedTime,
    expirationTime: state.app.expirationTime
  }
}

const mapDispatchToProps = (dispatch) => {
  return { actions: bindActionCreators(Object.assign({}, uiActions), dispatch) }
}

export class SessionMonitor extends Component {
  componentDidMount () {
    this.checkTimeout()
  }

  closeModal () {
    const { actions } = this.props
    actions.closeModal()
  }

  checkTimeout () {
    let self = this
    const { t, actions, loggedTime, expirationTime, sessionExpiringWarning, sessionExpiredReload, checkInterval } = this.props
    if (!loggedTime || !sessionExpiringWarning || !sessionExpiredReload || !checkInterval) {
      return
    }
    return setTimeout(() => {
      const now = new Date()
      const diff = now.getTime() - loggedTime.getTime()
      if (diff > sessionExpiredReload) {
        window.location.reload()
      } else {
        if (diff > sessionExpiringWarning) {
          const minutesRemaining = Math.abs(Math.ceil((expirationTime.getTime() - now.getTime()) / 1000 / 60))
          actions.openModal({
            modalTitle: t('ui:session-expire-title'),
            modalText: t('ui:session-expire-text', { minutes: minutesRemaining }),
            modalButtons: [{
              main: true,
              text: t('ui:ok-got-it'),
              onClick: this.closeModal.bind(this)
            }]
          })
        }
        self.checkTimeout()
      }
    }, checkInterval)
  }

  render () {
    return <div />
  }
}

SessionMonitor.propTypes = {
  loggedTime: PT.object,
  expirationTime: PT.object,
  sessionExpiringWarning: PT.number,
  sessionExpiredReload: PT.number,
  checkInterval: PT.number
}

const ConnectedSessionMonitor = connect(
  mapStateToProps,
  mapDispatchToProps
)(
  withTranslation()(SessionMonitor)
)

export default ConnectedSessionMonitor
