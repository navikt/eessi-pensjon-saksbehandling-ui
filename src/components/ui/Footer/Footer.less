.c-ui-footer {
    flex-shrink: 0;
    background: whitesmoke;
    padding: 0rem;
    display: flex;
    flex-direction: row;
    justify-content: space-between;
    transition: width 0.2s ease-out;

    &.toggled {
      height: 40px;
    }

    &:not(.toggled) {
      height: 10px;
    }

    .footerButtonClosed {
       width: 100%;
       margin: 0px;
       display: block;
       padding: 0px;
       &:hover {
         cursor: pointer;
         background-color: lightgrey;
       }
    }

    .footerButtonOpen {
       width: 1.2rem;
       margin-right: 0.5rem;
       display: inline-block;
       padding-top: 0.6rem;
       color: grey;
       text-align: center;
       &:hover {
         cursor: pointer;
         background-color: lightgrey;
       }
    }
    .fullWidth {
        width: 100%;
    }
    .contents {
      display:flex;
    }
    .newParam {
      transform: scale(0.7);
      transform-origin: left top;
      display: flex;
      flex-direction: row;
      align-items: baseline;
    }
    .paramSelect,
    .paramValue {
       margin-right: 0.5rem;
       margin-bottom: 0px;
    }
    .addParamButton, .toggleDrawerEnableButton {
      padding: 0.5rem;
    }
    .params {
      display: flex;
      flex-direction: row-reverse;

      div {
        margin-left: 0.25rem;
      }
    }

    .mini {
        transform: scale(0.5);
    }

    .param {
        display: flex;
        align-items: center;
        flex-direction: row;

        .etikett {
           padding: 0.2rem !important;
           white-space: nowrap;
        }
    }
}
