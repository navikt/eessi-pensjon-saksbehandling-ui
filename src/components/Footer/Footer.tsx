import { setStatusParam, unsetStatusParam } from 'actions/app'
import { toggleFooterOpen } from 'actions/ui'
import BUCWebSocket from 'applications/BUC/websocket/WebSocket'
import classNames from 'classnames'
import { HighContrastKnapp } from 'components/StyledComponents'
import { Person } from 'declarations/types'
import _ from 'lodash'
import EtikettBase from 'nav-frontend-etiketter'
import Knapp from 'nav-frontend-knapper'
import Lukknapp from 'nav-frontend-lukknapp'
import { Input, Select } from 'nav-frontend-skjema'
import { theme, themeHighContrast } from 'nav-styled-component-theme'
import PT from 'prop-types'
import React, { useState } from 'react'
import { useDispatch, useSelector } from 'react-redux'
import { Store } from 'redux'
import styled, { ThemeProvider } from 'styled-components'

export interface FooterProps {
  className ?: string
  highContrast: boolean
  footerOpen: boolean
  params: {[k: string]: any}
  person: Person | undefined
}

const FooterDiv = styled.footer`
  flex-shrink: 0;
  background-color: ${({ theme }: any) => theme['main-background-other-color']};
  padding: 0rem;
  display: flex;
  flex-direction: row;
  justify-content: space-between;
  transition: width 0.2s ease-out;
  &.toggled {
    height: 40px;
  }
  &:not(.toggled) {
    height: 10px;
  }
`
const ContentDiv = styled.div`
  display: flex;
  &.fullWidth {
    width: 100%;
  }
  .footerButtonClosed {
    width: 100%;
    margin: 0px;
    display: block;
    padding: 0px;
    background-color: lightgray;
  }
  .footerButtonClosed:hover {
    cursor: pointer;
    background-color: ${({ theme }: any) => theme['main-background-other-color']};
  }
  .footerButtonOpen {
    width: 1.2rem;
    margin-right: 0.5rem;
    display: inline-block;
    padding-top: 0.6rem;
    color: ${({ theme }: any) => theme['main-font-color']};
    text-align: center;
  }
  .footerButtonOpen:hover {
    cursor: pointer;
    background-color: ${({ theme }: any) => theme['main-background-other-color']};
  }
`
const FormDiv = styled.div`
  transform: scale(0.7);
  transform-origin: left top;
  display: flex;
  flex-direction: row;
  align-items: baseline;
`
const FooterSelect = styled(Select)`
  margin-right: 0.5rem;
  margin-bottom: 0px;
`
const FooterInput = styled(Input)`
  margin-right: 0.5rem;
  margin-bottom: 0px;
`
const AddButton = styled(Knapp)`
 padding: 0.5rem;
`

const ParamsDiv = styled.div`
  display: flex;
  flex-direction: row-reverse;

  div {
    margin-left: 0.25rem;
  }
`
const ParamDiv = styled.div`
  display: flex;
  align-items: center;
  flex-direction: row;

  .etikett {
    padding: 0.2rem !important;
    white-space: nowrap;
  }
`
const RemoveButton = styled(Lukknapp)`
  transform: scale(0.5);
`

const Footer: React.FC<FooterProps> = ({
  className, footerOpen, highContrast, params, person
}: FooterProps): JSX.Element => {
  const validParams: Array<string> = ['buc', 'sed', 'rinaId', 'sakId', 'aktoerId', 'vedtakId', 'kravId', 'fnr', 'mottaker']
  const [paramName, setParamName] = useState<string |undefined>(undefined)
  const [paramValue, setParamValue] = useState<string |undefined>(undefined)
  const store = useSelector<Store, Store>(state => state)
  const dispatch = useDispatch()
  const onUnsetParam = (key: string) => {
    dispatch(unsetStatusParam(key))
  }

  const onSetParam = () => {
    dispatch(setStatusParam(paramName!, paramValue))
    setParamName(undefined)
    setParamValue(undefined)
  }

  const onSetParamName = (e: React.ChangeEvent<HTMLSelectElement>) => {
    const name = e.target.value
    if (name !== '') {
      setParamName(name)
    }
  }

  const onSetParamValue = (e: React.ChangeEvent<HTMLInputElement>) => {
    const value = e.target.value
    if (value !== '') {
      setParamValue(value)
    }
  }

  const dumpStore = () => {
    let element = document.createElement('a')
    element.setAttribute('href', 'data:text/plain;charset=utf-8,' + encodeURIComponent(JSON.stringify(store, null, 2)))
    element.setAttribute('download', 'store.txt')
    element.style.display = 'none'
    document.body.appendChild(element)
    element.click()
    document.body.removeChild(element)
  }

  const _toggleFooterOpen = () => {
    dispatch(toggleFooterOpen())
  }

  return (
    <ThemeProvider theme={highContrast ? themeHighContrast : theme}>
      <FooterDiv role='contentinfo' className={classNames(className, { toggled: footerOpen })}>
        <ContentDiv className={classNames({ fullWidth: !footerOpen })}>
          <div
            className={classNames({ footerButtonOpen: footerOpen, footerButtonClosed: !footerOpen })}
            onClick={_toggleFooterOpen}
          >
            {footerOpen ? 'â–¼' : null}
          </div>
          {footerOpen && (
            <FormDiv>
              <FooterSelect
                date-testid='c-footer__select-id'
                label=''
                onChange={onSetParamName}
              >
                <option value=''>--</option>
                {validParams.map(param => {
                  return params[param] ? null : <option key={param} value={param}>{param}</option>
                })}
              </FooterSelect>
              <FooterInput
                label=''
                date-testid='c-footer__input-id'
                value={paramValue || ''}
                onChange={onSetParamValue}
              />
              <AddButton
                data-testid='c-footer__add-button-id'
                onClick={onSetParam}
              >
                &nbsp;+&nbsp;
              </AddButton>
              <HighContrastKnapp onClick={dumpStore}>
                Dump store
              </HighContrastKnapp>
            </FormDiv>
          )}
        </ContentDiv>
        {footerOpen && (
          <ParamsDiv>
            {validParams.map(param => {
              return params[param] && (
                <ParamDiv key={param}>
                  <EtikettBase className='c-footer__param-string' type='info'>
                    <b>{param}</b> {params[param]}
                  </EtikettBase>
                  <RemoveButton
                    onClick={() => onUnsetParam(param)}
                  />
                </ParamDiv>
              )
            })}
            <BUCWebSocket
              fnr={_.get(person, 'aktoer.ident.ident')}
              avdodfnr=''
            />
          </ParamsDiv>
        )}
      </FooterDiv>
    </ThemeProvider>
  )
}

Footer.propTypes = {
  className: PT.string,
  footerOpen: PT.bool.isRequired,
  params: PT.object.isRequired
}

export default Footer
