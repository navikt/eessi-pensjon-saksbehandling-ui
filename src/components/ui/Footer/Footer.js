import React, { Component } from 'react'
import PT from 'prop-types'
import { connect, bindActionCreators } from 'store'
import classNames from 'classnames'

import * as uiActions from '../../../actions/ui'
import * as appActions from '../../../actions/app'
import * as statusActions from '../../../actions/status'
import * as Nav from '../Nav'

import * as constants from '../../../constants/constants'

import './Footer.css'

const mapStateToProps = (state) => {
  return {
    userRole: state.app.userRole,
    rinaId: state.status.rinaId,
    sakId: state.status.sakId,
    aktoerId: state.status.aktoerId,
    vedtakId: state.status.vedtakId,
    kravId: state.status.kravId,
    fnr: state.status.fnr,
    sed: state.status.sed,
    buc: state.status.buc,
    mottaker: state.status.mottaker,
    footerOpen: state.ui.footerOpen
  }
}

const mapDispatchToProps = (dispatch) => {
  return { actions: bindActionCreators(Object.assign({}, appActions, uiActions, statusActions), dispatch) }
}

const params = ['buc', 'sed', 'rinaId', 'sakId', 'aktoerId', 'vedtakId', 'kravId', 'fnr', 'mottaker']

export class Footer extends Component {
  state = {}

  onUnsetParam (key) {
    const { actions } = this.props

    actions.unsetStatusParam(key)
  }

  onLogoClick () {
    const { actions, userRole } = this.props

    if (userRole === constants.SAKSBEHANDLER) {
      actions.toggleDrawerEnable()
    }
  }

  onSetParam () {
    const { actions } = this.props
    const { paramName, paramValue } = this.state

    actions.setStatusParam(paramName, paramValue)
    this.setState({
      paramName: undefined,
      paramValue: undefined
    })
  }

  onSetParamName (e) {
    let name = e.target ? e.target.value : e
    if (name !== '') {
      this.setState({
        paramName: name
      })
    }
  }

  onSetParamValue (e) {
    let value = e.target.value
    this.setState({
      paramValue: value
    })
  }

  toggleFooterOpen () {
    const { actions } = this.props
    actions.toggleFooterOpen()
  }

  render () {
    const { footerOpen } = this.props
    const { paramValue } = this.state

    return <footer className={classNames('c-ui-footer', { toggled: footerOpen })}>
      <div className={classNames('contents', { fullWidth: !footerOpen })}>
        <div className={classNames({ footerButtonOpen: footerOpen, footerButtonClosed: !footerOpen })}
          onClick={this.toggleFooterOpen.bind(this)}>
          {footerOpen ? 'â–¼' : null}
        </div>
        {footerOpen ? <div className='newParam'>
          <Nav.Select className='paramSelect' label='' onChange={this.onSetParamName.bind(this)}>
            <option value=''>{'--'}</option>
            {params.map(param => {
              return this.props[param] ? null : <option key={param} value={param}>{param}</option>
            })}
          </Nav.Select>
          <Nav.Input label='' className={'paramValue'} value={paramValue || ''}
            onChange={this.onSetParamValue.bind(this)} />
          <Nav.Knapp className='addParamButton' onClick={this.onSetParam.bind(this)}>&nbsp;+&nbsp;</Nav.Knapp>
          <Nav.Knapp className='toggleDrawerEnableButton' onClick={this.onLogoClick.bind(this)}>&nbsp;&lt;&lt;&nbsp;</Nav.Knapp>
        </div> : null}
      </div>
      {footerOpen ? <div className='params'>
        {params.map(param => {
          return this.props[param] ? <div key={param} className='param'>
            <Nav.EtikettBase type='info'><b>{param}</b> {this.props[param]}</Nav.EtikettBase>
            <Nav.Lukknapp className='mini' bla onClick={this.onUnsetParam.bind(this, param)} /></div> : null
        })}
      </div> : null}
    </footer>
  }
}

Footer.propTypes = {
  sed: PT.string,
  buc: PT.string,
  rinaId: PT.string,
  sakId: PT.string,
  aktoerId: PT.string,
  vedtakId: PT.string,
  kravId: PT.string,
  fnr: PT.string
}

export default connect(
  mapStateToProps,
  mapDispatchToProps
)(Footer)
