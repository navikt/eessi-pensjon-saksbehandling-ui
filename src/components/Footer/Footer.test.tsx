import { setStatusParam, unsetStatusParam } from 'actions/app'
import { toggleFooterOpen } from 'actions/ui'
import { mount, ReactWrapper } from 'enzyme'
import { stageSelector } from 'setupTests'
import Footer, { FooterProps } from './Footer'

jest.mock('actions/ui', () => ({
  toggleFooterOpen: jest.fn()
}))
jest.mock('actions/app', () => ({
  setStatusParam: jest.fn(),
  unsetStatusParam: jest.fn()
}))
describe('components/Footer', () => {
  let wrapper: ReactWrapper
  const initialMockProps: FooterProps = {
    footerOpen: true,
    params: {}
  }

  beforeEach(() => {
    stageSelector({}, {})
  })

  it('Render: match snapshot', () => {
    wrapper = mount(<Footer {...initialMockProps} />)
    expect(wrapper.isEmptyRender()).toBeFalsy()
    expect(wrapper).toMatchSnapshot()
  })

  it('Handling: Toggles open/closed with props', () => {
    wrapper = mount(<Footer {...initialMockProps} footerOpen={false} />)
    expect(wrapper.exists('.footerButtonClosed')).toBeTruthy()
    wrapper.setProps({ footerOpen: true })
    expect(wrapper.exists('.footerButtonOpen')).toBeTruthy()
  })

  it('Handling: Toggles open/closed with click', () => {
    (toggleFooterOpen as jest.Mock).mockReset()
    wrapper = mount(<Footer {...initialMockProps} footerOpen={false} />)
    wrapper.find('.footerButtonClosed').hostNodes().simulate('click')
    expect(toggleFooterOpen).toHaveBeenCalled()
  })

  it('Handling: Adds a param', () => {
    (setStatusParam as jest.Mock).mockReset()
    wrapper = mount(<Footer {...initialMockProps} />)
    wrapper.find('[data-test-id=\'c-footer__select-id\']').hostNodes().simulate('change', { target: { value: 'aktoerId' } })
    wrapper.find('[data-test-id=\'c-footer__input-id\']').hostNodes().simulate('change', { target: { value: '123' } })
    wrapper.find('[data-test-id=\'c-footer__add-button-id\']').hostNodes().simulate('click')
    expect(setStatusParam).toHaveBeenCalled()
  })

  it('Handling: Remove a param', () => {
    (unsetStatusParam as jest.Mock).mockReset()
    wrapper = mount(<Footer {...initialMockProps} params={{ sakId: '123' }} />)
    expect(wrapper.find('[data-test-id=\'c-footer__param-string\']').hostNodes().render().text()).toEqual('sakId 123')
    wrapper.find('[data-test-id=\'c-footer__remove-button\']').hostNodes().simulate('click')
    expect(unsetStatusParam).toHaveBeenCalled()
  })
})
