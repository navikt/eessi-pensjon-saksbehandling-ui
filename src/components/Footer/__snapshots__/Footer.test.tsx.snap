// Jest Snapshot v1, https://goo.gl/fbAQLP

exports['components/Footer Renders 1'] = `
<Footer
  footerOpen={true}
  params={Object {}}
>
  <footer
    className="c-footer toggled"
    role="contentinfo"
  >
    <div
      className="contents"
    >
      <div
        className="footerButtonOpen"
        onClick={[Function]}
      >
        ▼
      </div>
      <div
        className="c-footer__form"
      >
        <Select
          bredde="fullbredde"
          className="c-footer__select"
          disabled={false}
          id="c-footer__select-id"
          label=""
          onChange={[Function]}
        >
          <div
            className="skjemaelement c-footer__select"
          >
            <div
              className="selectContainer input--fullbredde"
            >
              <select
                aria-invalid={false}
                className="skjemaelement__input"
                disabled={false}
                id="c-footer__select-id"
                onChange={[Function]}
              >
                <option
                  value=""
                >
                  --
                </option>
                <option
                  key="buc"
                  value="buc"
                >
                  buc
                </option>
                <option
                  key="sed"
                  value="sed"
                >
                  sed
                </option>
                <option
                  key="rinaId"
                  value="rinaId"
                >
                  rinaId
                </option>
                <option
                  key="sakId"
                  value="sakId"
                >
                  sakId
                </option>
                <option
                  key="aktoerId"
                  value="aktoerId"
                >
                  aktoerId
                </option>
                <option
                  key="avdodfnr"
                  value="avdodfnr"
                >
                  avdodfnr
                </option>
                <option
                  key="vedtakId"
                  value="vedtakId"
                >
                  vedtakId
                </option>
                <option
                  key="kravId"
                  value="kravId"
                >
                  kravId
                </option>
                <option
                  key="fnr"
                  value="fnr"
                >
                  fnr
                </option>
                <option
                  key="mottaker"
                  value="mottaker"
                >
                  mottaker
                </option>
              </select>
            </div>
          </div>
        </Select>
        <Input
          bredde="fullbredde"
          className="c-footer__input"
          id="c-footer__input-id"
          label=""
          onChange={[Function]}
          value=""
        >
          <div
            className="skjemaelement c-footer__input"
          >
            <input
              aria-invalid={false}
              className="skjemaelement__input input--fullbredde"
              id="c-footer__input-id"
              onChange={[Function]}
              type="text"
              value=""
            />
          </div>
        </Input>
        <Knapp
          className="c-footer__add-button"
          id="c-footer__add-button-id"
          onClick={[Function]}
        >
          <KnappBase
            autoDisableVedSpinner={false}
            className="c-footer__add-button"
            disabled={false}
            htmlType="submit"
            id="c-footer__add-button-id"
            inaktivKlasseVedDisabled={false}
            kompakt={false}
            mini={false}
            onClick={[Function]}
            spinner={false}
            type="standard"
          >
            <button
              className="knapp c-footer__add-button knapp--standard"
              disabled={false}
              id="c-footer__add-button-id"
              onClick={[Function]}
              type="submit"
            >
               + 
            </button>
          </KnappBase>
        </Knapp>
      </div>
    </div>
    <div
      className="c-footer__params"
    >
      <Hovedknapp
        onClick={[Function]}
        size={10}
      >
        <KnappBase
          autoDisableVedSpinner={false}
          className={null}
          disabled={false}
          htmlType="submit"
          inaktivKlasseVedDisabled={false}
          kompakt={false}
          mini={false}
          onClick={[Function]}
          size={10}
          spinner={false}
          type="hoved"
        >
          <button
            className="knapp knapp--hoved"
            disabled={false}
            onClick={[Function]}
            size={10}
            type="submit"
          >
            Do Not Click
          </button>
        </KnappBase>
      </Hovedknapp>
    </div>
  </footer>
</Footer>
`
